### PR TITLE
Fix #6537: Bump BraveCore to 1.46.137

### LINKED ISSUE #6537: Turn off/on state isn't retained for the P3A switch after the browser restart
### Description: 
Turn off/on state isn't retained for the P3A switch after the browser restart

### Steps to Reproduce 

Case 1:

  1. Clean profile 1.46 (22.12.1.21)
  2. Open settings
  3. Navigate to `Brave Shields & Privacy`
  4. Scroll down to Privacy settings
  5. Turn off the `Allow privacy-preserving product analysis (P3A)` switch
  6. Restart the browser (remove app from memory and relaunch)
  7. P3A switch is turned ON after the browser restart
 

Case 2:

  1. Clean profile 1.46 (22.12.1.21)
  2. Ensured P3A modal is shown in the onbrading after `Set Default` card/slide
  3. Turn off the `Show anonymus, private product insights` switch
  4. Navigate to `Brave Shields & Privacy`
  5. Scroll down to Privacy settings
  6. Ensured `Allow privacy-preserving product analysis (P3A)` switch is OFF
  7. Turn on the `Allow privacy-preserving product analysis (P3A)` switch
  8. Restart the browser (remove app from memory and relaunch)
  9. P3A switch is turned OFF after the browser restart
   
Case 3:

  1. Clean profile 1.45.x
  2. Upgrade profile to 1.46 (22.12.1.21)
  3. Ensured P3A modal is shown on the 1st launch in an upgraded profile
  4. Turn off the `Show anonymus, private product insights` switch
  5. Go to `Brave Shields & Privacy`
  6. Scroll down to Privacy settings
  7. Ensured `Allow privacy-preserving product analysis (P3A)` switch is OFF
  10. Turn ON the switch `Allow privacy-preserving product analysis (P3A)` 
  11. Restart the browser (remove app from memory and relaunch)
  12. P3A switch is turned OFF after the browser restart

**Actual result:** 
Turn off/on state isn't retained for the P3A switch after the browser restart

**Expected result:**
Turn off/on state should be retained for the P3A switch after the browser restart

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 

1.46 (22.12.1.21)
**Device details:** 

iPhone 8 - iOS 16.1.1
**Website problems only:**
- did you check with Brave Shields down?NA
- did you check in Safari/Firefox (WkWebView-based browsers)? NA

### Additional Information

cc : @brave/qa-team @kylehickinson @anthonypkeane

### PR DESCRIPTION
**Pull request by kylehickinson**
## Summary of Changes

This pull request fixes #6537 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by StephenHeaps — APPROVED** [review]

File: Client/Frontend/Browser/Toolbars/UrlBar/CollapsedURLBarView.swift [hunk 1]
```diff
@@ -68,7 +68,7 @@ class CollapsedURLBarView: UIView {
   var currentURL: URL? {
     didSet {
       urlLabel.text = currentURL.map {
-        URLFormatter.formatURL(forDisplayOmitSchemePathAndTrivialSubdomains: $0.absoluteString)
+        URLFormatter.formatURLOrigin(forDisplayOmitSchemePathAndTrivialSubdomains: $0.absoluteString)
```
Comment (soner-yuksel) [inline Client/Frontend/Browser/Toolbars/UrlBar/CollapsedURLBarView.swift:71]: Wondering If we should add `formatURLOrigin` to 
https://github.com/brave/brave-ios/blob/5611ad2babaaa848ad0bd873ca2c00c255cfb27f/Client/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift#L411

and maybe edit locationText

https://github.com/brave/brave-ios/blob/5611ad2babaaa848ad0bd873ca2c00c255cfb27f/Client/Frontend/Browser/Toolbars/UrlBar/TopToolbarView.swift#LL666C9-L666C9

Comment (kylehickinson) [reply Client/Frontend/Browser/Toolbars/UrlBar/CollapsedURLBarView.swift:71]: Lets do a separate PR for that

Comment (kylehickinson) [reply Client/Frontend/Browser/Toolbars/UrlBar/CollapsedURLBarView.swift:71]: Plus the API for that is going to be the new one (`formatURL`) not `formatURLOrigin`

Comment (soner-yuksel) [reply Client/Frontend/Browser/Toolbars/UrlBar/CollapsedURLBarView.swift:71]: Sounds good, I will check in detail after we merge core update

**Review by soner-yuksel — APPROVED** [review]

**Pull request merged**